qt/python/mantidqt/widgets/memorywidget/memorypresenter.py

```diff
@@ -25,18 +25,33 @@ class MemoryPresenter(object):
     """
     def __init__(self, view):
         self.view = view
-        self.timer = QTimer()
+        self.timer = Timer(TIME_INTERVAL_MEMORY_USAGE_UPDATE, self.update_memory_usage)
+        self.thread_on = False
+        self.updating_cancelled = False
+        self.application_closing = False
         self.update_memory_usage()
         self.set_bar_color_at_start()
-        self.update_at_regular_intervals()
+        self.start_memory_widget_update_thread()
 
-    def update_at_regular_intervals(self):
+    def __del__(self):
+        self.cancel_memory_update()
+
+    def start_memory_widget_update_thread(self):
         """
         Sets timer so that the memory usage is updated
         every TIME_INTERVAL_MEMORY_USAGE_UPDATE (ms)
         """
-        self.timer.timeout.connect(self.update_memory_usage)
-        self.timer.start(TIME_INTERVAL_MEMORY_USAGE_UPDATE)
+        if not self.thread_on:
+            self.timer.start()
+
+    def _spin_off_another_time_thread(self):
+        """
+        Spins off another timer thread, by creating a new Timer thread object and starting it
+        """
+        if not self.thread_on and not self.application_closing:
+            self.timer = Timer(TIME_INTERVAL_MEMORY_USAGE_UPDATE, self.update_memory_usage)
```

Comment:
`Timer` is a subclass of Thread so this means that we start a new thread every ~2s? That's quite expensive. Can be do something bit like the interval code [here](https://stackoverflow.com/q/5179467) so we only need to start 1 thread and this thread simply calls the function repeatedly?

Reply:
I have implemented based on that stackoverflow discussion you linked, it "should" be clean and work in other scenarios so I stuck it into mantidqt.utils